sql/core/src/test/resources/sql-tests/inputs/pgSQL/aggregates_part2.sql

```diff
@@ -0,0 +1,217 @@
+     -- null because strict
+--   boolor_statefunc(NULL, NULL)  IS NULL AS "t",
+--   boolor_statefunc(TRUE, NULL)  IS NULL AS "t",
+--   boolor_statefunc(FALSE, NULL) IS NULL AS "t",
+--   boolor_statefunc(NULL, TRUE)  IS NULL AS "t",
+--   boolor_statefunc(NULL, FALSE) IS NULL AS "t",
+     -- actual computations
+--   boolor_statefunc(TRUE, TRUE) AS "t",
+--   boolor_statefunc(TRUE, FALSE) AS "t",
+--   boolor_statefunc(FALSE, TRUE) AS "t",
+--   NOT boolor_statefunc(FALSE, FALSE) AS "t";
+
+-- [SPARK-27880] Implement boolean aggregates(BOOL_AND, BOOL_OR and EVERY)
+-- CREATE TEMPORARY TABLE bool_test(
+--   b1 BOOL,
+--   b2 BOOL,
+--   b3 BOOL,
+--   b4 BOOL);
+
+-- empty case
+-- SELECT
+--   BOOL_AND(b1)   AS "n",
+--   BOOL_OR(b3)    AS "n"
+-- FROM bool_test;
+
+-- COPY bool_test FROM STDIN NULL 'null';
+-- TRUE	null	FALSE	null
+-- FALSE	TRUE	null	null
+-- null	TRUE	FALSE	null
+-- \.
+
+-- SELECT
+--   BOOL_AND(b1)     AS "f",
+--   BOOL_AND(b2)     AS "t",
+--   BOOL_AND(b3)     AS "f",
+--   BOOL_AND(b4)     AS "n",
+--   BOOL_AND(NOT b2) AS "f",
+--   BOOL_AND(NOT b3) AS "t"
+-- FROM bool_test;
+
+-- SELECT
+--   EVERY(b1)     AS "f",
+--   EVERY(b2)     AS "t",
+--   EVERY(b3)     AS "f",
+--   EVERY(b4)     AS "n",
+--   EVERY(NOT b2) AS "f",
+--   EVERY(NOT b3) AS "t"
+-- FROM bool_test;
+
+-- SELECT
+--   BOOL_OR(b1)      AS "t",
+--   BOOL_OR(b2)      AS "t",
+--   BOOL_OR(b3)      AS "f",
+--   BOOL_OR(b4)      AS "n",
+--   BOOL_OR(NOT b2)  AS "f",
+--   BOOL_OR(NOT b3)  AS "t"
+-- FROM bool_test;
+
+--
+-- Test cases that should be optimized into indexscans instead of
+-- the generic aggregate implementation.
+--
+
+-- Basic cases
+explain
+  select min(unique1) from tenk1;
+select min(unique1) from tenk1;
+explain
+  select max(unique1) from tenk1;
+select max(unique1) from tenk1;
+explain
+  select max(unique1) from tenk1 where unique1 < 42;
+select max(unique1) from tenk1 where unique1 < 42;
+explain
+  select max(unique1) from tenk1 where unique1 > 42;
+select max(unique1) from tenk1 where unique1 > 42;
+
+-- the planner may choose a generic aggregate here if parallel query is
+-- enabled, since that plan will be parallel safe and the "optimized"
+-- plan, which has almost identical cost, will not be.  we want to test
+-- the optimized plan, so temporarily disable parallel query.
+-- begin;
+-- set local max_parallel_workers_per_gather = 0;
+explain
+  select max(unique1) from tenk1 where unique1 > 42000;
+select max(unique1) from tenk1 where unique1 > 42000;
+-- rollback;
+
+-- multi-column index (uses tenk1_thous_tenthous)
+explain
+  select max(tenthous) from tenk1 where thousand = 33;
+select max(tenthous) from tenk1 where thousand = 33;
+explain
+  select min(tenthous) from tenk1 where thousand = 33;
+select min(tenthous) from tenk1 where thousand = 33;
+
+-- check parameter propagation into an indexscan subquery
+explain
+  select f1, (select min(unique1) from tenk1 where unique1 > f1) AS gt
+    from int4_tbl;
```

Comment:
How to create `int4_tbl`? `int4_tbl` is not created by [copy.source](https://github.com/postgres/postgres/blob/REL_12_BETA1/src/test/regress/input/copy.source), via [int4.sql#L5-L18](https://github.com/postgres/postgres/blob/REL_12_BETA1/src/test/regress/sql/int4.sql#L5-L18)?

Reply:
You can try to create a table at the beginning of this test suite and compare the results. If that matched, I think that is the right source. 

You also can create a temp view for that 
```
create temporary view int4_tbl as select * from values
  (...),
...
  (...)
  as int4_tbl(f1);
```

Reply:
Done